base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -172,18 +172,10 @@ impl LMDBDatabase {
             trace!(target: LOG_TARGET, "[apply_db_transaction] WriteOperation: {}", op);
             match op {
                 InsertOrphanBlock(block) => self.insert_orphan_block(&write_txn, &block)?,
-                Delete(delete) => self.op_delete(&write_txn, delete)?,
                 InsertHeader { header } => {
-                    let height = header.header.height;
-                    if !self.insert_header(&write_txn, &header.header, &header.accumulated_data)? {
```

Comment:
Yes, the check has been moved into the method and will return the error there. The method now returns `Result<(),...>` instead of `Result<bool, ...>` and errors when an existing header exists